src/main.ts

```diff
@@ -1,491 +1,535 @@
-    );
+    this.hooks = {
+      beforeRun: new SyncHook(['config']),
+      getAuthor: new AsyncSeriesBailHook([]),
+      getPreviousVersion: new AsyncSeriesBailHook(['prefixRelease']),
+      getRepository: new AsyncSeriesBailHook([]),
+      publish: new AsyncSeriesHook(['version'])
+    };
   }
 
-  if (lastRelease.match(/\d+\.\d+\.\d+/)) {
-    veryVerbose.info('Using latest release as previous version');
-    lastVersion = lastRelease;
-  } else if (monorepoVersion) {
-    veryVerbose.info('Using lerna.json as previous version');
-    lastVersion = monorepoVersion;
-  } else if (packageVersion) {
-    veryVerbose.info('Using package.json as previous version');
-    lastVersion = packageVersion;
-  } else {
-    veryVerbose.info('No previous release found, using 0.0.0 as the start.');
-    lastVersion = prefixRelease('0.0.0');
-  }
+  public async loadConfig() {
+    const explorer = cosmiconfig('auto');
+    const result = await explorer.search();
 
-  // This helps in situations where the latest release on github is wrong for some reason
-  // In this case we default to either the monorepo or package version
-  lastVersion = calcGreaterVersion(
-    packageVersion,
-    monorepoVersion,
-    lastRelease
-  );
+    let rawConfig: cosmiconfig.Config = {};
 
-  return lastVersion;
-}
+    if (result && result.config) {
+      rawConfig = result.config;
+    }
 
-async function setGitUser(args: IGitHubReleaseOptions) {
-  try {
-    // If these values are not set git config will exit with an error
-    await execPromise(`git config user.email`);
-    await execPromise(`git config user.name`);
-  } catch (error) {
-    const packageJson = JSON.parse(await readFile('package.json', 'utf-8'));
-    let { name, email } = args;
+    this.logger.verbose.success(
+      'Loaded `auto-release` with config:',
+      rawConfig
+    );
 
-    if (!name && packageJson.author) {
-      ({ name } = packageJson.author);
-    }
+    this.semVerLabels = defaultLabels;
 
-    if (!email && packageJson.author) {
-      ({ email } = packageJson.author);
+    if (rawConfig.labels) {
+      this.semVerLabels = {
+        ...defaultLabels,
+        ...rawConfig.labels
+      };
     }
 
-    if (email) {
-      await execPromise(`git config user.email "${email}"`);
-    }
+    this.logger.verbose.success(
+      'Using SEMVER labels:',
+      '\n',
+      this.semVerLabels
+    );
 
-    if (name) {
-      await execPromise(`git config user.name "${name}"`);
+    const skipReleaseLabels = rawConfig.skipReleaseLabels || [];
+
+    if (
+      this.semVerLabels &&
+      !skipReleaseLabels.includes(this.semVerLabels.get('skip-release')!)
+    ) {
+      skipReleaseLabels.push(this.semVerLabels.get('skip-release')!);
     }
-  }
-}
 
-async function getVersion(githubRelease: GitHubRelease, args: ArgsType) {
-  const lastRelease = await githubRelease.getLatestRelease();
+    this.plugins.map(plugin => {
```

Comment:
Doesn't look like you need the `map` here, just looping over plugins

```suggestion
    this.plugins.forEach(plugin => {
```